polygeist/tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -2504,12 +2507,10 @@ void MLIRASTConsumer::setMLIRFunctionAttributes(FunctionOpInterface Function,
   mlirclang::AttributeList PAL;
   {
     const clang::CodeGen::CGFunctionInfo &FI = getOrCreateCGFunctionInfo(&FD);
-    const auto *FPT = FD.getType()->getAs<clang::FunctionProtoType>();
-    clang::CodeGen::CGCalleeInfo CalleeInfo(FPT);
 
     unsigned CallingConv;
-    getTypes().constructAttributeList(Function.getName(), FI, CalleeInfo, PAL,
-                                      CallingConv,
+    getTypes().constructAttributeList(Function.getName(), FI,
+                                      getGlobalDecl(&FD), PAL, CallingConv,
```

Comment:
`const Decl *TargetDecl = CalleeInfo.getCalleeDecl().getDecl()` is empty with the original `CalleeInfo`, so attributes that rely on `TargetDecl` were missed.
Changed to create `CalleeInfo` using `getGlobalDecl(&FD)`.